### PR TITLE
Remove tons of generated code overrides

### PR DESCRIPTION
**Pull request by christothes**
Updated the swagger reference to the latest and greatest which eliminated the need for much of the generated code overrides.

### CODE REVIEW

**Review by pakrym — APPROVED** [review]

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs [hunk 1]
```diff
@@ -89,11 +90,17 @@ public virtual async Task<Response<TableItem>> CreateAsync(CancellationToken can
...
-                                                     queryOptions: new QueryOptions() { Format = _format },
-                                                     cancellationToken: cancellationToken).ConfigureAwait(false);
+        public virtual async Task<Response<ReadOnlyDictionary<string, object>>> InsertAsync(IDictionary<string, object> entity, CancellationToken cancellationToken = default)
+        {
+            var response = await _tableOperations.InsertEntityAsync(_table,
+                                                                     tableEntityProperties: entity.ToOdataAnnotatedDictionary(),
+                                                                     queryOptions: new QueryOptions() { Format = _format },
+                                                                     cancellationToken: cancellationToken).ConfigureAwait(false);
+
+            ((IDictionary<string, object>)response.Value).CastAndRemoveAnnotations();
```
Comment (pakrym) [inline sdk/tables/Azure.Data.Tables/src/TableClient.cs:100]: Can we make a copy instead?

File: sdk/tables/Azure.Data.Tables/src/Extensions/DictionaryTableExtensions.cs [hunk 2]
```diff
@@ -93,6 +93,44 @@ internal static void CastAndRemoveAnnotations(this IReadOnlyList<IDictionary<str
...
+        /// Cleans a Dictionary of its Odata type annotations, while using them to cast its entities accordingly.
+        /// </summary>
+        internal static void CastAndRemoveAnnotations(this IDictionary<string, object> entity)
+        {
+            var typeAnnotationsWithKeys = new Dictionary<string, (string typeAnnotation, string annotationKey)>();
+            var spanOdataSuffix = TableConstants.Odata.OdataTypeString.AsSpan();
+
+            typeAnnotationsWithKeys.Clear();
+
+            foreach (var propertyName in entity.Keys)
```
Comment (pakrym) [inline sdk/tables/Azure.Data.Tables/src/Extensions/DictionaryTableExtensions.cs:106]: There is a bit of duplication with other CastAndRemoveAnnotations but might not matter yet.

Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/Extensions/DictionaryTableExtensions.cs:106]: Yes, it is mostly dupe, but it got tricky to convert into a reusable method so I left it this way for now.

**Comment by christothes** [comment]
fixes #11951

**Pull request merged**